mqtt/mqtt-bridge/src/persist/waking_state/waking_map.rs

```diff
@@ -0,0 +1,56 @@
+use std::{cmp::min, collections::HashMap, collections::VecDeque, task::Waker};
+
+use mqtt3::proto::Publication;
+
+use crate::persist::{waking_state::StreamWakeableState, Key, PersistError};
+
+/// Responsible for waking waiting streams when new elements are added.
+///
+/// Exposes a get method for retrieving a count of elements in order of insertion.
+/// When elements are retrieved they are moved to the in flight collection.
+pub struct WakingMap {
+    queue: VecDeque<(Key, Publication)>,
+    in_flight: HashMap<Key, Publication>,
+    waker: Option<Waker>,
```

Comment:
I think it shouldn't be possible to create a WakingMap without Waker. Why is it an option?

Reply:
A waker is set when a stream needs to be waked. When the waking map is created, then at that point there won't be any waiting streams, and thus no waker.